{
 "metadata": {
  "name": ""
 },
 "nbformat": 3,
 "nbformat_minor": 0,
 "worksheets": [
  {
   "cells": [
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "from bcbio import isatab"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 3
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "#rec= isatab.parse('/Users/abhishek/Downloads/isa_7283/')\n",
      "rec= isatab.parse('/Users/abhishek/projects/isa_tab/data/ISA-tab_Munoz-Heck_July26')\n",
      "#rec = isatab.parse('/Users/abhishek/projects/isa_tab/data/BII-S-3')"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 4
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "print rec.studies"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "[<bcbio.isatab.parser.ISATabStudyRecord instance at 0x1036399e0>]\n"
       ]
      }
     ],
     "prompt_number": 5
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "print 'ISA-TAB Summary'\n",
      "print '#studies %d' % len(rec.studies)\n",
      "print '#assays %d' % sum([len(study.assays) for study in rec.studies])"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "ISA-TAB Summary\n",
        "#studies 1\n",
        "#assays 2\n"
       ]
      }
     ],
     "prompt_number": 6
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "for study in rec.studies:\n",
      "    print 'Study:---------------'\n",
      "    print dir(study)\n",
      "    \n",
      "    \"\"\"\n",
      "    for assay in study.assays:\n",
      "        print assay.metadata\n",
      "    \"\"\""
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "Study:---------------\n",
        "['__doc__', '__init__', '__module__', '__str__', 'assays', 'contacts', 'design_descriptors', 'factors', 'metadata', 'nodes', 'protocols', 'publications']\n"
       ]
      }
     ],
     "prompt_number": 7
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "study.contacts"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 8,
       "text": [
        "[{'# Investigation File generated for Scientific Data with Investigation File Authoring Tool version 0.1.0 on 26/07/2013': '',\n",
        "  'Comment [Funder Term Accession Number]': '',\n",
        "  'Comment [Funder Term Source REF]': '',\n",
        "  'Comment [Funder]': '',\n",
        "  'Comment [Grant Identifier]': '',\n",
        "  'Comment [Study Person ORCID]': '',\n",
        "  'Study Person Address': '',\n",
        "  'Study Person Affiliation': 'Netherlands Proteomics Centre, Padualaan 8, Utrecht, 3584 CH, Netherlands',\n",
        "  'Study Person Email': 'A.J.R.Heck@uu.nl',\n",
        "  'Study Person Fax': '',\n",
        "  'Study Person First Name': 'Albert',\n",
        "  'Study Person Last Name': 'Heck',\n",
        "  'Study Person Mid Initials': '',\n",
        "  'Study Person Phone': '',\n",
        "  'Study Person Roles': '',\n",
        "  'Study Person Roles Term Accession Number': '',\n",
        "  'Study Person Roles Term Source REF': ''},\n",
        " {'# Investigation File generated for Scientific Data with Investigation File Authoring Tool version 0.1.0 on 26/07/2013': '',\n",
        "  'Comment [Funder Term Accession Number]': '',\n",
        "  'Comment [Funder Term Source REF]': '',\n",
        "  'Comment [Funder]': '',\n",
        "  'Comment [Grant Identifier]': '',\n",
        "  'Comment [Study Person ORCID]': '',\n",
        "  'Study Person Address': '',\n",
        "  'Study Person Affiliation': 'Bijvoet Center for Biomolecular Research and Utrecht Institute for Pharmaceutical Sciences, Utrecht University, Padualaan 8, Utrecht, 3584 CH, Netherlands',\n",
        "  'Study Person Email': '',\n",
        "  'Study Person Fax': '',\n",
        "  'Study Person First Name': 'Javier',\n",
        "  'Study Person Last Name': 'Munoz',\n",
        "  'Study Person Mid Initials': '',\n",
        "  'Study Person Phone': '',\n",
        "  'Study Person Roles': '',\n",
        "  'Study Person Roles Term Accession Number': '',\n",
        "  'Study Person Roles Term Source REF': ''}]"
       ]
      }
     ],
     "prompt_number": 8
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "# Study Tite\n",
      "print 'Study Title %s \\n\\n' % study.metadata['Study Title']\n",
      "\n",
      "\n",
      "#Study Description\n",
      "print 'Study Description %s \\n\\n' % study.metadata['Study Description']\n",
      "\n",
      "\n",
      "#study submission date\n",
      "print 'Study Submission Date %s \\n\\n ' % study.metadata['Study Submission Date']\n",
      "\n",
      "\n",
      "#study public release date\n",
      "print 'Study Public Release Date %s ' % study.metadata['Study Public Release Date']\n"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "Study Title Proteomic Profiles of Human Embryonic Stem Cells, Induced-Pluripotent Stem Cells and Precursor Fibroblasts \n",
        "\n",
        "\n",
        "Study Description  \n",
        "\n",
        "\n",
        "Study Submission Date  \n",
        "\n",
        " \n",
        "Study Public Release Date  \n"
       ]
      }
     ],
     "prompt_number": 9
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "#1. Study Design Type \n",
      "'   '.join([ descp['Study Design Type'] for descp in study.design_descriptors ])"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 10,
       "text": [
        "'cell type comparison design   growth condition intervention design'"
       ]
      }
     ],
     "prompt_number": 10
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "dir(study)"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 11,
       "text": [
        "['__doc__',\n",
        " '__init__',\n",
        " '__module__',\n",
        " '__str__',\n",
        " 'assays',\n",
        " 'contacts',\n",
        " 'design_descriptors',\n",
        " 'factors',\n",
        " 'metadata',\n",
        " 'nodes',\n",
        " 'protocols',\n",
        " 'publications']"
       ]
      }
     ],
     "prompt_number": 11
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "#contacts\n",
      "#working on following structure\n",
      "\n",
      "'''\n",
      "{'Study Person Address': 'Prospect Place, Plymouth, United Kingdom',\n",
      "  'Study Person Affiliation': 'Plymouth Marine Laboratory',\n",
      "  'Study Person Email': 'jagi@pml.ac.uk',\n",
      "  'Study Person Fax': '',\n",
      "  'Study Person First Name': 'Jack',\n",
      "  'Study Person Last Name': 'Gilbert',\n",
      "  'Study Person Mid Initials': 'A',\n",
      "  'Study Person Phone': '',\n",
      "  'Study Person Roles': 'principal investigator role',\n",
      "  'Study Person Roles Term Accession Number': '103',\n",
      "  'Study Person Roles Term Source REF': 'OBI'},\n",
      "'''\n",
      "\n",
      "full_names = []\n",
      "for contact in study.contacts:\n",
      "    full_name = '%s %s %s' % ( contact.get('Study Person First Name','NA'),\n",
      "                               contact.get('Study Person Mid Initials','NA'),\n",
      "                               contact.get('Study Person Last Name'))\n",
      "    full_names.append(full_name)\n",
      "\n",
      "print ', '.join(full_names)"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "Albert  Heck, Javier  Munoz\n"
       ]
      }
     ],
     "prompt_number": 12
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "study.publications"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 13,
       "text": [
        "[{'Study PubMed ID': 'PMID: 22108792',\n",
        "  'Study Publication Author List': 'Munoz J, Low TY, Kok YJ, Chin A, Frese CK, Ding V, Choo A, Heck AJ',\n",
        "  'Study Publication DOI': 'doi: 10.1038/msb.2011.84',\n",
        "  'Study Publication Status': 'published',\n",
        "  'Study Publication Status Term Accession Number': '',\n",
        "  'Study Publication Status Term Source REF': '',\n",
        "  'Study Publication Title': 'The quantitative proteomes of human-induced pluripotent stem cells and embryonic stem cells'}]"
       ]
      }
     ],
     "prompt_number": 13
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "#publications\n",
      "publications_line = ''\n",
      "for publication in study.publications:\n",
      "    publications_line += publication['Study Publication Author list'] + '\\n'\n",
      "    publications_line += publication['Study Publication Title']\n",
      "    publications_line += '[Pubmed](http://www.ncbi.nlm.nih.gov/pubmed/%s) \\n\\n' % publication.get('Study PubMed ID','NA')\n",
      "\n",
      "print publications_line"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "ename": "KeyError",
       "evalue": "'Study Publication Author list'",
       "output_type": "pyerr",
       "traceback": [
        "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m\n\u001b[0;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
        "\u001b[0;32m<ipython-input-14-7f32aac0de91>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[1;32m      2\u001b[0m \u001b[0mpublications_line\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m''\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      3\u001b[0m \u001b[0;32mfor\u001b[0m \u001b[0mpublication\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mstudy\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mpublications\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 4\u001b[0;31m     \u001b[0mpublications_line\u001b[0m \u001b[0;34m+=\u001b[0m \u001b[0mpublication\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'Study Publication Author list'\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m+\u001b[0m \u001b[0;34m'\\n'\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      5\u001b[0m     \u001b[0mpublications_line\u001b[0m \u001b[0;34m+=\u001b[0m \u001b[0mpublication\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'Study Publication Title'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      6\u001b[0m     \u001b[0mpublications_line\u001b[0m \u001b[0;34m+=\u001b[0m \u001b[0;34m'[Pubmed](http://www.ncbi.nlm.nih.gov/pubmed/%s) \\n\\n'\u001b[0m \u001b[0;34m%\u001b[0m \u001b[0mpublication\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'Study PubMed ID'\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m'NA'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;31mKeyError\u001b[0m: 'Study Publication Author list'"
       ]
      }
     ],
     "prompt_number": 14
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "for protocol in study.protocols:\n",
      "    print protocol.get('Study Protocol Name','NA')\n",
      "    print '%s' % protocol.get('Study Protocol Description', 'NA')"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "Cell Culture\n",
        "\n",
        "Flow cytometry analysis (FACS)\n",
        "\n",
        "Staining of hESC for markers\n",
        "\n",
        "Karyotypic stability\n",
        "\n",
        "In-vivo differentiation assay, SCID mice model and teratoma analysis\n",
        "\n",
        "Sample preparation for MS\n",
        "\n",
        "Mass spectrometric analysis\n",
        "\n",
        "Data processing\n",
        "\n",
        "RNA isolation\n",
        "\n",
        "Transcriptome analysis\n",
        "\n",
        "Microarray data analysis\n",
        "\n"
       ]
      }
     ],
     "prompt_number": 15
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "for assay in study.assays:\n",
      "    print dir(assay)\n",
      "    for key,val in assay.nodes.iteritems():\n",
      "        print key\n",
      "    print '----------'"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "['__doc__', '__init__', '__module__', '__str__', 'metadata', 'nodes']\n",
        "ftp://ftp.pride.ebi.ac.uk/2013/02/PXD000134/OR1_091015_Javier_iPS_Singapore_Mix1_BigSCX*.raw\n",
        "ftp://ftp.pride.ebi.ac.uk/2013/02/PXD000134/4Skin_Mix2_SCX*.raw\n",
        "ftp://ftp.pride.ebi.ac.uk/2013/02/PXD000134/4Skin_Mix1_SCX*.raw\n",
        "ftp://ftp.pride.ebi.ac.uk/2013/02/PXD000134/OR1_091015_Javier_iPS_Singapore_Mix2_BigSCX*.raw\n",
        "----------\n",
        "['__doc__', '__init__', '__module__', '__str__', 'metadata', 'nodes']\n",
        "http://www.ncbi.nlm.nih.gov/geo/download/?acc=GSM649323&format=file&file=GSM649323%2ECEL%2Egz\n",
        "http://www.ncbi.nlm.nih.gov/geo/download/?acc=GSM649336&format=file&file=GSM649336%2ECEL%2Egz\n",
        "http://www.ncbi.nlm.nih.gov/geo/download/?acc=GSM649335&format=file&file=GSM649335%2ECEL%2Egz\n",
        "http://www.ncbi.nlm.nih.gov/geo/download/?acc=GSM649337&format=file&file=GSM649337%2ECEL%2Egz\n",
        "http://www.ncbi.nlm.nih.gov/geo/download/?acc=GSM649332&format=file&file=GSM649332%2ECEL%2Egz\n",
        "http://www.ncbi.nlm.nih.gov/geo/download/?acc=GSM649324&format=file&file=GSM649324%2ECEL%2Egz\n",
        "http://www.ncbi.nlm.nih.gov/geo/download/?acc=GSM649321&format=file&file=GSM649321%2ECEL%2Egz\n",
        "http://www.ncbi.nlm.nih.gov/geo/download/?acc=GSM649333&format=file&file=GSM649333%2ECEL%2Egz\n",
        "http://www.ncbi.nlm.nih.gov/geo/download/?acc=GSM649320&format=file&file=GSM649320%2ECEL%2Egz\n",
        "http://www.ncbi.nlm.nih.gov/geo/download/?acc=GSM649319&format=file&file=GSM649319%2ECEL%2Egz\n",
        "http://www.ncbi.nlm.nih.gov/geo/download/?acc=GSM649322&format=file&file=GSM649322%2ECEL%2Egz\n",
        "http://www.ncbi.nlm.nih.gov/geo/download/?acc=GSM649334&format=file&file=GSM649334%2ECEL%2Egz\n",
        "----------\n"
       ]
      }
     ],
     "prompt_number": 16
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "print assay"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "    * Assay\n",
        "     metadata: {'Study Assay File Name': 'a_assay_SCIDATA-example1-2.txt',\n",
        "        'Study Assay Measurement Type': 'transcription profiling',\n",
        "        'Study Assay Measurement Type Term Accession Number': '',\n",
        "        'Study Assay Measurement Type Term Source REF': '',\n",
        "        'Study Assay Technology Platform': 'Affymetrix U133 Plus 2.0 GeneChip Microarrays',\n",
        "        'Study Assay Technology Type': 'DNA microarray',\n",
        "        'Study Assay Technology Type Term Accession Number': '',\n",
        "        'Study Assay Technology Type Term Source REF': ''}\n",
        "     nodes:\n",
        "       * Node http://www.ncbi.nlm.nih.gov/geo/download/?acc=GSM649323&format=file&file=GSM649323%2ECEL%2Egz Raw Data File\n",
        "         metadata: {'Data Record Accession': [Attrs(Data_Record_Accession='GSE26451'),\n",
        "                                    Attrs(Data_Record_Accession='')],\n",
        "          'Data Repository': [Attrs(Data_Repository=''),\n",
        "                              Attrs(Data_Repository='Gene Expression Omnibus')],\n",
        "          'Derived Data File': [''],\n",
        "          'Label': [Attrs(Label='')],\n",
        "          'Labeled Extract Name': [Attrs(Labeled_Extract_Name='')],\n",
        "          'Protocol REF': [Attrs(Protocol_REF='Microarray data analysis'),\n",
        "                           Attrs(Protocol_REF=''),\n",
        "                           Attrs(Protocol_REF='Transcriptome analysis'),\n",
        "                           Attrs(Protocol_REF='RNA isolation')],\n",
        "          'Raw Data File': ['http://www.ncbi.nlm.nih.gov/geo/download/?acc=GSM649323&format=file&file=GSM649323%2ECEL%2Egz'],\n",
        "          'Sample Name': ['4Skin_Fibro_1']}\n",
        "       * Node http://www.ncbi.nlm.nih.gov/geo/download/?acc=GSM649336&format=file&file=GSM649336%2ECEL%2Egz Raw Data File\n",
        "         metadata: {'Data Record Accession': [Attrs(Data_Record_Accession='GSE26453'),\n",
        "                                    Attrs(Data_Record_Accession='')],\n",
        "          'Data Repository': [Attrs(Data_Repository=''),\n",
        "                              Attrs(Data_Repository='Gene Expression Omnibus')],\n",
        "          'Derived Data File': [''],\n",
        "          'Label': [Attrs(Label='')],\n",
        "          'Labeled Extract Name': [Attrs(Labeled_Extract_Name='')],\n",
        "          'Protocol REF': [Attrs(Protocol_REF='Microarray data analysis'),\n",
        "                           Attrs(Protocol_REF=''),\n",
        "                           Attrs(Protocol_REF='Transcriptome analysis'),\n",
        "                           Attrs(Protocol_REF='RNA isolation')],\n",
        "          'Raw Data File': ['http://www.ncbi.nlm.nih.gov/geo/download/?acc=GSM649336&format=file&file=GSM649336%2ECEL%2Egz'],\n",
        "          'Sample Name': ['IMR90_Fibro_1']}\n",
        "       * Node http://www.ncbi.nlm.nih.gov/geo/download/?acc=GSM649335&format=file&file=GSM649335%2ECEL%2Egz Raw Data File\n",
        "         metadata: {'Data Record Accession': [Attrs(Data_Record_Accession='GSE26453'),\n",
        "                                    Attrs(Data_Record_Accession='')],\n",
        "          'Data Repository': [Attrs(Data_Repository=''),\n",
        "                              Attrs(Data_Repository='Gene Expression Omnibus')],\n",
        "          'Derived Data File': [''],\n",
        "          'Label': [Attrs(Label='')],\n",
        "          'Labeled Extract Name': [Attrs(Labeled_Extract_Name='')],\n",
        "          'Protocol REF': [Attrs(Protocol_REF='Microarray data analysis'),\n",
        "                           Attrs(Protocol_REF=''),\n",
        "                           Attrs(Protocol_REF='Transcriptome analysis'),\n",
        "                           Attrs(Protocol_REF='RNA isolation')],\n",
        "          'Raw Data File': ['http://www.ncbi.nlm.nih.gov/geo/download/?acc=GSM649335&format=file&file=GSM649335%2ECEL%2Egz'],\n",
        "          'Sample Name': ['HES-3_2']}\n",
        "       * Node http://www.ncbi.nlm.nih.gov/geo/download/?acc=GSM649337&format=file&file=GSM649337%2ECEL%2Egz Raw Data File\n",
        "         metadata: {'Data Record Accession': [Attrs(Data_Record_Accession='GSE26453'),\n",
        "                                    Attrs(Data_Record_Accession='')],\n",
        "          'Data Repository': [Attrs(Data_Repository=''),\n",
        "                              Attrs(Data_Repository='Gene Expression Omnibus')],\n",
        "          'Derived Data File': [''],\n",
        "          'Label': [Attrs(Label='')],\n",
        "          'Labeled Extract Name': [Attrs(Labeled_Extract_Name='')],\n",
        "          'Protocol REF': [Attrs(Protocol_REF='Microarray data analysis'),\n",
        "                           Attrs(Protocol_REF=''),\n",
        "                           Attrs(Protocol_REF='Transcriptome analysis'),\n",
        "                           Attrs(Protocol_REF='RNA isolation')],\n",
        "          'Raw Data File': ['http://www.ncbi.nlm.nih.gov/geo/download/?acc=GSM649337&format=file&file=GSM649337%2ECEL%2Egz'],\n",
        "          'Sample Name': ['IMR90_Fibro_2']}\n",
        "       * Node http://www.ncbi.nlm.nih.gov/geo/download/?acc=GSM649332&format=file&file=GSM649332%2ECEL%2Egz Raw Data File\n",
        "         metadata: {'Data Record Accession': [Attrs(Data_Record_Accession='GSE26453'),\n",
        "                                    Attrs(Data_Record_Accession='')],\n",
        "          'Data Repository': [Attrs(Data_Repository=''),\n",
        "                              Attrs(Data_Repository='Gene Expression Omnibus')],\n",
        "          'Derived Data File': [''],\n",
        "          'Label': [Attrs(Label='')],\n",
        "          'Labeled Extract Name': [Attrs(Labeled_Extract_Name='')],\n",
        "          'Protocol REF': [Attrs(Protocol_REF='Microarray data analysis'),\n",
        "                           Attrs(Protocol_REF=''),\n",
        "                           Attrs(Protocol_REF='Transcriptome analysis'),\n",
        "                           Attrs(Protocol_REF='RNA isolation')],\n",
        "          'Raw Data File': ['http://www.ncbi.nlm.nih.gov/geo/download/?acc=GSM649332&format=file&file=GSM649332%2ECEL%2Egz'],\n",
        "          'Sample Name': ['IMR90_iPS_1']}\n",
        "       * Node http://www.ncbi.nlm.nih.gov/geo/download/?acc=GSM649324&format=file&file=GSM649324%2ECEL%2Egz Raw Data File\n",
        "         metadata: {'Data Record Accession': [Attrs(Data_Record_Accession='GSE26451'),\n",
        "                                    Attrs(Data_Record_Accession='')],\n",
        "          'Data Repository': [Attrs(Data_Repository=''),\n",
        "                              Attrs(Data_Repository='Gene Expression Omnibus')],\n",
        "          'Derived Data File': [''],\n",
        "          'Label': [Attrs(Label='')],\n",
        "          'Labeled Extract Name': [Attrs(Labeled_Extract_Name='')],\n",
        "          'Protocol REF': [Attrs(Protocol_REF='Microarray data analysis'),\n",
        "                           Attrs(Protocol_REF=''),\n",
        "                           Attrs(Protocol_REF='Transcriptome analysis'),\n",
        "                           Attrs(Protocol_REF='RNA isolation')],\n",
        "          'Raw Data File': ['http://www.ncbi.nlm.nih.gov/geo/download/?acc=GSM649324&format=file&file=GSM649324%2ECEL%2Egz'],\n",
        "          'Sample Name': ['4Skin_Fibro_2']}\n",
        "       * Node http://www.ncbi.nlm.nih.gov/geo/download/?acc=GSM649321&format=file&file=GSM649321%2ECEL%2Egz Raw Data File\n",
        "         metadata: {'Data Record Accession': [Attrs(Data_Record_Accession='GSE26451'),\n",
        "                                    Attrs(Data_Record_Accession='')],\n",
        "          'Data Repository': [Attrs(Data_Repository=''),\n",
        "                              Attrs(Data_Repository='Gene Expression Omnibus')],\n",
        "          'Derived Data File': [''],\n",
        "          'Label': [Attrs(Label='')],\n",
        "          'Labeled Extract Name': [Attrs(Labeled_Extract_Name='')],\n",
        "          'Protocol REF': [Attrs(Protocol_REF='Microarray data analysis'),\n",
        "                           Attrs(Protocol_REF=''),\n",
        "                           Attrs(Protocol_REF='Transcriptome analysis'),\n",
        "                           Attrs(Protocol_REF='RNA isolation')],\n",
        "          'Raw Data File': ['http://www.ncbi.nlm.nih.gov/geo/download/?acc=GSM649321&format=file&file=GSM649321%2ECEL%2Egz'],\n",
        "          'Sample Name': ['HES-3_3']}\n",
        "       * Node http://www.ncbi.nlm.nih.gov/geo/download/?acc=GSM649333&format=file&file=GSM649333%2ECEL%2Egz Raw Data File\n",
        "         metadata: {'Data Record Accession': [Attrs(Data_Record_Accession='GSE26453'),\n",
        "                                    Attrs(Data_Record_Accession='')],\n",
        "          'Data Repository': [Attrs(Data_Repository=''),\n",
        "                              Attrs(Data_Repository='Gene Expression Omnibus')],\n",
        "          'Derived Data File': [''],\n",
        "          'Label': [Attrs(Label='')],\n",
        "          'Labeled Extract Name': [Attrs(Labeled_Extract_Name='')],\n",
        "          'Protocol REF': [Attrs(Protocol_REF='Microarray data analysis'),\n",
        "                           Attrs(Protocol_REF=''),\n",
        "                           Attrs(Protocol_REF='Transcriptome analysis'),\n",
        "                           Attrs(Protocol_REF='RNA isolation')],\n",
        "          'Raw Data File': ['http://www.ncbi.nlm.nih.gov/geo/download/?acc=GSM649333&format=file&file=GSM649333%2ECEL%2Egz'],\n",
        "          'Sample Name': ['IMR90_iPS_2']}\n",
        "       * Node http://www.ncbi.nlm.nih.gov/geo/download/?acc=GSM649320&format=file&file=GSM649320%2ECEL%2Egz Raw Data File\n",
        "         metadata: {'Data Record Accession': [Attrs(Data_Record_Accession='GSE26451'),\n",
        "                                    Attrs(Data_Record_Accession='')],\n",
        "          'Data Repository': [Attrs(Data_Repository=''),\n",
        "                              Attrs(Data_Repository='Gene Expression Omnibus')],\n",
        "          'Derived Data File': [''],\n",
        "          'Label': [Attrs(Label='')],\n",
        "          'Labeled Extract Name': [Attrs(Labeled_Extract_Name='')],\n",
        "          'Protocol REF': [Attrs(Protocol_REF='Microarray data analysis'),\n",
        "                           Attrs(Protocol_REF=''),\n",
        "                           Attrs(Protocol_REF='Transcriptome analysis'),\n",
        "                           Attrs(Protocol_REF='RNA isolation')],\n",
        "          'Raw Data File': ['http://www.ncbi.nlm.nih.gov/geo/download/?acc=GSM649320&format=file&file=GSM649320%2ECEL%2Egz'],\n",
        "          'Sample Name': ['4Skin_iPS_2']}\n",
        "       * Node http://www.ncbi.nlm.nih.gov/geo/download/?acc=GSM649319&format=file&file=GSM649319%2ECEL%2Egz Raw Data File\n",
        "         metadata: {'Data Record Accession': [Attrs(Data_Record_Accession='GSE26451'),\n",
        "                                    Attrs(Data_Record_Accession='')],\n",
        "          'Data Repository': [Attrs(Data_Repository=''),\n",
        "                              Attrs(Data_Repository='Gene Expression Omnibus')],\n",
        "          'Derived Data File': [''],\n",
        "          'Label': [Attrs(Label='')],\n",
        "          'Labeled Extract Name': [Attrs(Labeled_Extract_Name='')],\n",
        "          'Protocol REF': [Attrs(Protocol_REF='Microarray data analysis'),\n",
        "                           Attrs(Protocol_REF=''),\n",
        "                           Attrs(Protocol_REF='Transcriptome analysis'),\n",
        "                           Attrs(Protocol_REF='RNA isolation')],\n",
        "          'Raw Data File': ['http://www.ncbi.nlm.nih.gov/geo/download/?acc=GSM649319&format=file&file=GSM649319%2ECEL%2Egz'],\n",
        "          'Sample Name': ['4Skin_iPS_1']}\n",
        "       * Node http://www.ncbi.nlm.nih.gov/geo/download/?acc=GSM649322&format=file&file=GSM649322%2ECEL%2Egz Raw Data File\n",
        "         metadata: {'Data Record Accession': [Attrs(Data_Record_Accession='GSE26451'),\n",
        "                                    Attrs(Data_Record_Accession='')],\n",
        "          'Data Repository': [Attrs(Data_Repository=''),\n",
        "                              Attrs(Data_Repository='Gene Expression Omnibus')],\n",
        "          'Derived Data File': [''],\n",
        "          'Label': [Attrs(Label='')],\n",
        "          'Labeled Extract Name': [Attrs(Labeled_Extract_Name='')],\n",
        "          'Protocol REF': [Attrs(Protocol_REF='Microarray data analysis'),\n",
        "                           Attrs(Protocol_REF=''),\n",
        "                           Attrs(Protocol_REF='Transcriptome analysis'),\n",
        "                           Attrs(Protocol_REF='RNA isolation')],\n",
        "          'Raw Data File': ['http://www.ncbi.nlm.nih.gov/geo/download/?acc=GSM649322&format=file&file=GSM649322%2ECEL%2Egz'],\n",
        "          'Sample Name': ['HES-3_4']}\n",
        "       * Node http://www.ncbi.nlm.nih.gov/geo/download/?acc=GSM649334&format=file&file=GSM649334%2ECEL%2Egz Raw Data File\n",
        "         metadata: {'Data Record Accession': [Attrs(Data_Record_Accession='GSE26453'),\n",
        "                                    Attrs(Data_Record_Accession='')],\n",
        "          'Data Repository': [Attrs(Data_Repository=''),\n",
        "                              Attrs(Data_Repository='Gene Expression Omnibus')],\n",
        "          'Derived Data File': [''],\n",
        "          'Label': [Attrs(Label='')],\n",
        "          'Labeled Extract Name': [Attrs(Labeled_Extract_Name='')],\n",
        "          'Protocol REF': [Attrs(Protocol_REF='Microarray data analysis'),\n",
        "                           Attrs(Protocol_REF=''),\n",
        "                           Attrs(Protocol_REF='Transcriptome analysis'),\n",
        "                           Attrs(Protocol_REF='RNA isolation')],\n",
        "          'Raw Data File': ['http://www.ncbi.nlm.nih.gov/geo/download/?acc=GSM649334&format=file&file=GSM649334%2ECEL%2Egz'],\n",
        "          'Sample Name': ['HES-3_1']}\n",
        "\n"
       ]
      }
     ],
     "prompt_number": 17
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "dir(assay)"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 15,
       "text": [
        "['__doc__', '__init__', '__module__', '__str__', 'metadata', 'nodes']"
       ]
      }
     ],
     "prompt_number": 15
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "assay.metadata"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "metadata": {},
       "output_type": "pyout",
       "prompt_number": 16,
       "text": [
        "{'Study Assay File Name': 'a_gilbert-assay-Tx.txt',\n",
        " 'Study Assay Measurement Type': 'transcription profiling',\n",
        " 'Study Assay Measurement Type Term Accession Number': '424',\n",
        " 'Study Assay Measurement Type Term Source REF': 'OBI',\n",
        " 'Study Assay Technology Platform': '454 Genome Sequencer FS',\n",
        " 'Study Assay Technology Type': 'nucleotide sequencing',\n",
        " 'Study Assay Technology Type Term Accession Number': '',\n",
        " 'Study Assay Technology Type Term Source REF': 'OBI'}"
       ]
      }
     ],
     "prompt_number": 16
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "for node in assay.nodes:\n",
      "    print assay.nodes.get(node,'NA')"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "output_type": "stream",
       "stream": "stdout",
       "text": [
        "       * Node EVHINN104.sff Raw Data File\n",
        "         metadata: {'Derived Data File': ['GSE10119_series_matrix.txt'],\n",
        "          'Material Type': [Attrs(Material_Type='mRNA', Term_Source_REF='CHEBI', Term_Accession_Number='16991')],\n",
        "          'Protocol REF': [Attrs(Protocol_REF='mRNA extraction - standard procedure 3'),\n",
        "                           Attrs(Protocol_REF='nucleic acid extraction - standard procedure 2'),\n",
        "                           Attrs(Protocol_REF='library construction'),\n",
        "                           Attrs(Protocol_REF='sequence analysis - standard procedure 7'),\n",
        "                           Attrs(Protocol_REF='reverse transcription - standard procedure 5'),\n",
        "                           Attrs(Protocol_REF='pyrosequencing - standard procedure 6')],\n",
        "          'Raw Data File': ['EVHINN104.sff'],\n",
        "          'Sample Name': ['GSM255770']}\n",
        "       * Node EVHINN115.sff Raw Data File\n",
        "         metadata: {'Derived Data File': ['GSE10119_series_matrix.txt'],\n",
        "          'Material Type': [Attrs(Material_Type='mRNA', Term_Source_REF='CHEBI', Term_Accession_Number='16991')],\n",
        "          'Protocol REF': [Attrs(Protocol_REF='mRNA extraction - standard procedure 3'),\n",
        "                           Attrs(Protocol_REF='nucleic acid extraction - standard procedure 2'),\n",
        "                           Attrs(Protocol_REF='library construction'),\n",
        "                           Attrs(Protocol_REF='sequence analysis - standard procedure 7'),\n",
        "                           Attrs(Protocol_REF='reverse transcription - standard procedure 5'),\n",
        "                           Attrs(Protocol_REF='pyrosequencing - standard procedure 6')],\n",
        "          'Raw Data File': ['EVHINN115.sff'],\n",
        "          'Sample Name': ['GSM255773']}\n",
        "       * Node EVHINN105.sff Raw Data File\n",
        "         metadata: {'Derived Data File': ['GSE10119_series_matrix.txt'],\n",
        "          'Material Type': [Attrs(Material_Type='mRNA', Term_Source_REF='CHEBI', Term_Accession_Number='16991')],\n",
        "          'Protocol REF': [Attrs(Protocol_REF='mRNA extraction - standard procedure 3'),\n",
        "                           Attrs(Protocol_REF='nucleic acid extraction - standard procedure 2'),\n",
        "                           Attrs(Protocol_REF='library construction'),\n",
        "                           Attrs(Protocol_REF='sequence analysis - standard procedure 7'),\n",
        "                           Attrs(Protocol_REF='reverse transcription - standard procedure 5'),\n",
        "                           Attrs(Protocol_REF='pyrosequencing - standard procedure 6')],\n",
        "          'Raw Data File': ['EVHINN105.sff'],\n",
        "          'Sample Name': ['GSM255771']}\n",
        "       * Node EVNG8PH03.sff Raw Data File\n",
        "         metadata: {'Derived Data File': ['GSE10119_series_matrix.txt'],\n",
        "          'Material Type': [Attrs(Material_Type='mRNA', Term_Source_REF='CHEBI', Term_Accession_Number='16991')],\n",
        "          'Protocol REF': [Attrs(Protocol_REF='mRNA extraction - standard procedure 3'),\n",
        "                           Attrs(Protocol_REF='nucleic acid extraction - standard procedure 2'),\n",
        "                           Attrs(Protocol_REF='library construction'),\n",
        "                           Attrs(Protocol_REF='sequence analysis - standard procedure 7'),\n",
        "                           Attrs(Protocol_REF='reverse transcription - standard procedure 5'),\n",
        "                           Attrs(Protocol_REF='pyrosequencing - standard procedure 6')],\n",
        "          'Raw Data File': ['EVNG8PH03.sff'],\n",
        "          'Sample Name': ['GSM255772']}\n",
        "       * Node EVUSNDQ01.sff Raw Data File\n",
        "         metadata: {'Derived Data File': ['GSE10119_series_matrix.txt'],\n",
        "          'Material Type': [Attrs(Material_Type='mRNA', Term_Source_REF='CHEBI', Term_Accession_Number='16991')],\n",
        "          'Protocol REF': [Attrs(Protocol_REF='mRNA extraction - standard procedure 3'),\n",
        "                           Attrs(Protocol_REF='nucleic acid extraction - standard procedure 2'),\n",
        "                           Attrs(Protocol_REF='library construction'),\n",
        "                           Attrs(Protocol_REF='sequence analysis - standard procedure 7'),\n",
        "                           Attrs(Protocol_REF='reverse transcription - standard procedure 5'),\n",
        "                           Attrs(Protocol_REF='pyrosequencing - standard procedure 6')],\n",
        "          'Raw Data File': ['EVUSNDQ01.sff'],\n",
        "          'Sample Name': ['GSM255770']}\n",
        "       * Node EVHINN101.sff Raw Data File\n",
        "         metadata: {'Derived Data File': ['GSE10119_series_matrix.txt'],\n",
        "          'Material Type': [Attrs(Material_Type='mRNA', Term_Source_REF='CHEBI', Term_Accession_Number='16991')],\n",
        "          'Protocol REF': [Attrs(Protocol_REF='mRNA extraction - standard procedure 3'),\n",
        "                           Attrs(Protocol_REF='nucleic acid extraction - standard procedure 2'),\n",
        "                           Attrs(Protocol_REF='library construction'),\n",
        "                           Attrs(Protocol_REF='sequence analysis - standard procedure 7'),\n",
        "                           Attrs(Protocol_REF='reverse transcription - standard procedure 5'),\n",
        "                           Attrs(Protocol_REF='pyrosequencing - standard procedure 6')],\n",
        "          'Raw Data File': ['EVHINN101.sff'],\n",
        "          'Sample Name': ['GSM255770']}\n",
        "       * Node EVHINN108.sff Raw Data File\n",
        "         metadata: {'Derived Data File': ['GSE10119_series_matrix.txt'],\n",
        "          'Material Type': [Attrs(Material_Type='mRNA', Term_Source_REF='CHEBI', Term_Accession_Number='16991')],\n",
        "          'Protocol REF': [Attrs(Protocol_REF='mRNA extraction - standard procedure 3'),\n",
        "                           Attrs(Protocol_REF='nucleic acid extraction - standard procedure 2'),\n",
        "                           Attrs(Protocol_REF='library construction'),\n",
        "                           Attrs(Protocol_REF='sequence analysis - standard procedure 7'),\n",
        "                           Attrs(Protocol_REF='reverse transcription - standard procedure 5'),\n",
        "                           Attrs(Protocol_REF='pyrosequencing - standard procedure 6')],\n",
        "          'Raw Data File': ['EVHINN108.sff'],\n",
        "          'Sample Name': ['GSM255771']}\n",
        "       * Node EVHINN103.sff Raw Data File\n",
        "         metadata: {'Derived Data File': ['GSE10119_series_matrix.txt'],\n",
        "          'Material Type': [Attrs(Material_Type='mRNA', Term_Source_REF='CHEBI', Term_Accession_Number='16991')],\n",
        "          'Protocol REF': [Attrs(Protocol_REF='mRNA extraction - standard procedure 3'),\n",
        "                           Attrs(Protocol_REF='nucleic acid extraction - standard procedure 2'),\n",
        "                           Attrs(Protocol_REF='library construction'),\n",
        "                           Attrs(Protocol_REF='sequence analysis - standard procedure 7'),\n",
        "                           Attrs(Protocol_REF='reverse transcription - standard procedure 5'),\n",
        "                           Attrs(Protocol_REF='pyrosequencing - standard procedure 6')],\n",
        "          'Raw Data File': ['EVHINN103.sff'],\n",
        "          'Sample Name': ['GSM255770']}\n",
        "       * Node EVNG8PH02.sff Raw Data File\n",
        "         metadata: {'Derived Data File': ['GSE10119_series_matrix.txt'],\n",
        "          'Material Type': [Attrs(Material_Type='mRNA', Term_Source_REF='CHEBI', Term_Accession_Number='16991')],\n",
        "          'Protocol REF': [Attrs(Protocol_REF='mRNA extraction - standard procedure 3'),\n",
        "                           Attrs(Protocol_REF='nucleic acid extraction - standard procedure 2'),\n",
        "                           Attrs(Protocol_REF='library construction'),\n",
        "                           Attrs(Protocol_REF='sequence analysis - standard procedure 7'),\n",
        "                           Attrs(Protocol_REF='reverse transcription - standard procedure 5'),\n",
        "                           Attrs(Protocol_REF='pyrosequencing - standard procedure 6')],\n",
        "          'Raw Data File': ['EVNG8PH02.sff'],\n",
        "          'Sample Name': ['GSM255771']}\n",
        "       * Node EVHINN111.sff Raw Data File\n",
        "         metadata: {'Derived Data File': ['GSE10119_series_matrix.txt'],\n",
        "          'Material Type': [Attrs(Material_Type='mRNA', Term_Source_REF='CHEBI', Term_Accession_Number='16991')],\n",
        "          'Protocol REF': [Attrs(Protocol_REF='mRNA extraction - standard procedure 3'),\n",
        "                           Attrs(Protocol_REF='nucleic acid extraction - standard procedure 2'),\n",
        "                           Attrs(Protocol_REF='library construction'),\n",
        "                           Attrs(Protocol_REF='sequence analysis - standard procedure 7'),\n",
        "                           Attrs(Protocol_REF='reverse transcription - standard procedure 5'),\n",
        "                           Attrs(Protocol_REF='pyrosequencing - standard procedure 6')],\n",
        "          'Raw Data File': ['EVHINN111.sff'],\n",
        "          'Sample Name': ['GSM255772']}\n",
        "       * Node EVNG8PH01.sff Raw Data File\n",
        "         metadata: {'Derived Data File': ['GSE10119_series_matrix.txt'],\n",
        "          'Material Type': [Attrs(Material_Type='mRNA', Term_Source_REF='CHEBI', Term_Accession_Number='16991')],\n",
        "          'Protocol REF': [Attrs(Protocol_REF='mRNA extraction - standard procedure 3'),\n",
        "                           Attrs(Protocol_REF='nucleic acid extraction - standard procedure 2'),\n",
        "                           Attrs(Protocol_REF='library construction'),\n",
        "                           Attrs(Protocol_REF='sequence analysis - standard procedure 7'),\n",
        "                           Attrs(Protocol_REF='reverse transcription - standard procedure 5'),\n",
        "                           Attrs(Protocol_REF='pyrosequencing - standard procedure 6')],\n",
        "          'Raw Data File': ['EVNG8PH01.sff'],\n",
        "          'Sample Name': ['GSM255770']}\n",
        "       * Node EVHINN110.sff Raw Data File\n",
        "         metadata: {'Derived Data File': ['GSE10119_series_matrix.txt'],\n",
        "          'Material Type': [Attrs(Material_Type='mRNA', Term_Source_REF='CHEBI', Term_Accession_Number='16991')],\n",
        "          'Protocol REF': [Attrs(Protocol_REF='mRNA extraction - standard procedure 3'),\n",
        "                           Attrs(Protocol_REF='nucleic acid extraction - standard procedure 2'),\n",
        "                           Attrs(Protocol_REF='library construction'),\n",
        "                           Attrs(Protocol_REF='sequence analysis - standard procedure 7'),\n",
        "                           Attrs(Protocol_REF='reverse transcription - standard procedure 5'),\n",
        "                           Attrs(Protocol_REF='pyrosequencing - standard procedure 6')],\n",
        "          'Raw Data File': ['EVHINN110.sff'],\n",
        "          'Sample Name': ['GSM255772']}\n",
        "       * Node EVUSNDQ04.sff Raw Data File\n",
        "         metadata: {'Derived Data File': ['GSE10119_series_matrix.txt'],\n",
        "          'Material Type': [Attrs(Material_Type='mRNA', Term_Source_REF='CHEBI', Term_Accession_Number='16991')],\n",
        "          'Protocol REF': [Attrs(Protocol_REF='mRNA extraction - standard procedure 3'),\n",
        "                           Attrs(Protocol_REF='nucleic acid extraction - standard procedure 2'),\n",
        "                           Attrs(Protocol_REF='library construction'),\n",
        "                           Attrs(Protocol_REF='sequence analysis - standard procedure 7'),\n",
        "                           Attrs(Protocol_REF='reverse transcription - standard procedure 5'),\n",
        "                           Attrs(Protocol_REF='pyrosequencing - standard procedure 6')],\n",
        "          'Raw Data File': ['EVUSNDQ04.sff'],\n",
        "          'Sample Name': ['GSM255773']}\n",
        "       * Node EVHINN113.sff Raw Data File\n",
        "         metadata: {'Derived Data File': ['GSE10119_series_matrix.txt'],\n",
        "          'Material Type': [Attrs(Material_Type='mRNA', Term_Source_REF='CHEBI', Term_Accession_Number='16991')],\n",
        "          'Protocol REF': [Attrs(Protocol_REF='mRNA extraction - standard procedure 3'),\n",
        "                           Attrs(Protocol_REF='nucleic acid extraction - standard procedure 2'),\n",
        "                           Attrs(Protocol_REF='library construction'),\n",
        "                           Attrs(Protocol_REF='sequence analysis - standard procedure 7'),\n",
        "                           Attrs(Protocol_REF='reverse transcription - standard procedure 5'),\n",
        "                           Attrs(Protocol_REF='pyrosequencing - standard procedure 6')],\n",
        "          'Raw Data File': ['EVHINN113.sff'],\n",
        "          'Sample Name': ['GSM255773']}\n",
        "       * Node EVHINN106.sff Raw Data File\n",
        "         metadata: {'Derived Data File': ['GSE10119_series_matrix.txt'],\n",
        "          'Material Type': [Attrs(Material_Type='mRNA', Term_Source_REF='CHEBI', Term_Accession_Number='16991')],\n",
        "          'Protocol REF': [Attrs(Protocol_REF='mRNA extraction - standard procedure 3'),\n",
        "                           Attrs(Protocol_REF='nucleic acid extraction - standard procedure 2'),\n",
        "                           Attrs(Protocol_REF='library construction'),\n",
        "                           Attrs(Protocol_REF='sequence analysis - standard procedure 7'),\n",
        "                           Attrs(Protocol_REF='reverse transcription - standard procedure 5'),\n",
        "                           Attrs(Protocol_REF='pyrosequencing - standard procedure 6')],\n",
        "          'Raw Data File': ['EVHINN106.sff'],\n",
        "          'Sample Name': ['GSM255771']}\n",
        "       * Node EVHINN109.sff Raw Data File\n",
        "         metadata: {'Derived Data File': ['GSE10119_series_matrix.txt'],\n",
        "          'Material Type': [Attrs(Material_Type='mRNA', Term_Source_REF='CHEBI', Term_Accession_Number='16991')],\n",
        "          'Protocol REF': [Attrs(Protocol_REF='mRNA extraction - standard procedure 3'),\n",
        "                           Attrs(Protocol_REF='nucleic acid extraction - standard procedure 2'),\n",
        "                           Attrs(Protocol_REF='library construction'),\n",
        "                           Attrs(Protocol_REF='sequence analysis - standard procedure 7'),\n",
        "                           Attrs(Protocol_REF='reverse transcription - standard procedure 5'),\n",
        "                           Attrs(Protocol_REF='pyrosequencing - standard procedure 6')],\n",
        "          'Raw Data File': ['EVHINN109.sff'],\n",
        "          'Sample Name': ['GSM255772']}\n",
        "       * Node EVHINN102.sff Raw Data File\n",
        "         metadata: {'Derived Data File': ['GSE10119_series_matrix.txt'],\n",
        "          'Material Type': [Attrs(Material_Type='mRNA', Term_Source_REF='CHEBI', Term_Accession_Number='16991')],\n",
        "          'Protocol REF': [Attrs(Protocol_REF='mRNA extraction - standard procedure 3'),\n",
        "                           Attrs(Protocol_REF='nucleic acid extraction - standard procedure 2'),\n",
        "                           Attrs(Protocol_REF='library construction'),\n",
        "                           Attrs(Protocol_REF='sequence analysis - standard procedure 7'),\n",
        "                           Attrs(Protocol_REF='reverse transcription - standard procedure 5'),\n",
        "                           Attrs(Protocol_REF='pyrosequencing - standard procedure 6')],\n",
        "          'Raw Data File': ['EVHINN102.sff'],\n",
        "          'Sample Name': ['GSM255770']}\n",
        "       * Node EVHINN112.sff Raw Data File\n",
        "         metadata: {'Derived Data File': ['GSE10119_series_matrix.txt'],\n",
        "          'Material Type': [Attrs(Material_Type='mRNA', Term_Source_REF='CHEBI', Term_Accession_Number='16991')],\n",
        "          'Protocol REF': [Attrs(Protocol_REF='mRNA extraction - standard procedure 3'),\n",
        "                           Attrs(Protocol_REF='nucleic acid extraction - standard procedure 2'),\n",
        "                           Attrs(Protocol_REF='library construction'),\n",
        "                           Attrs(Protocol_REF='sequence analysis - standard procedure 7'),\n",
        "                           Attrs(Protocol_REF='reverse transcription - standard procedure 5'),\n",
        "                           Attrs(Protocol_REF='pyrosequencing - standard procedure 6')],\n",
        "          'Raw Data File': ['EVHINN112.sff'],\n",
        "          'Sample Name': ['GSM255772']}\n",
        "       * Node EVHINN116.sff Raw Data File\n",
        "         metadata: {'Derived Data File': ['GSE10119_series_matrix.txt'],\n",
        "          'Material Type': [Attrs(Material_Type='mRNA', Term_Source_REF='CHEBI', Term_Accession_Number='16991')],\n",
        "          'Protocol REF': [Attrs(Protocol_REF='mRNA extraction - standard procedure 3'),\n",
        "                           Attrs(Protocol_REF='nucleic acid extraction - standard procedure 2'),\n",
        "                           Attrs(Protocol_REF='library construction'),\n",
        "                           Attrs(Protocol_REF='sequence analysis - standard procedure 7'),\n",
        "                           Attrs(Protocol_REF='reverse transcription - standard procedure 5'),\n",
        "                           Attrs(Protocol_REF='pyrosequencing - standard procedure 6')],\n",
        "          'Raw Data File': ['EVHINN116.sff'],\n",
        "          'Sample Name': ['GSM255773']}\n",
        "       * Node EVHINN114.sff Raw Data File\n",
        "         metadata: {'Derived Data File': ['GSE10119_series_matrix.txt'],\n",
        "          'Material Type': [Attrs(Material_Type='mRNA', Term_Source_REF='CHEBI', Term_Accession_Number='16991')],\n",
        "          'Protocol REF': [Attrs(Protocol_REF='mRNA extraction - standard procedure 3'),\n",
        "                           Attrs(Protocol_REF='nucleic acid extraction - standard procedure 2'),\n",
        "                           Attrs(Protocol_REF='library construction'),\n",
        "                           Attrs(Protocol_REF='sequence analysis - standard procedure 7'),\n",
        "                           Attrs(Protocol_REF='reverse transcription - standard procedure 5'),\n",
        "                           Attrs(Protocol_REF='pyrosequencing - standard procedure 6')],\n",
        "          'Raw Data File': ['EVHINN114.sff'],\n",
        "          'Sample Name': ['GSM255773']}\n",
        "       * Node EVHINN107.sff Raw Data File\n",
        "         metadata: {'Derived Data File': ['GSE10119_series_matrix.txt'],\n",
        "          'Material Type': [Attrs(Material_Type='mRNA', Term_Source_REF='CHEBI', Term_Accession_Number='16991')],\n",
        "          'Protocol REF': [Attrs(Protocol_REF='mRNA extraction - standard procedure 3'),\n",
        "                           Attrs(Protocol_REF='nucleic acid extraction - standard procedure 2'),\n",
        "                           Attrs(Protocol_REF='library construction'),\n",
        "                           Attrs(Protocol_REF='sequence analysis - standard procedure 7'),\n",
        "                           Attrs(Protocol_REF='reverse transcription - standard procedure 5'),\n",
        "                           Attrs(Protocol_REF='pyrosequencing - standard procedure 6')],\n",
        "          'Raw Data File': ['EVHINN107.sff'],\n",
        "          'Sample Name': ['GSM255771']}\n",
        "       * Node EVUSNDQ03.sff Raw Data File\n",
        "         metadata: {'Derived Data File': ['GSE10119_series_matrix.txt'],\n",
        "          'Material Type': [Attrs(Material_Type='mRNA', Term_Source_REF='CHEBI', Term_Accession_Number='16991')],\n",
        "          'Protocol REF': [Attrs(Protocol_REF='mRNA extraction - standard procedure 3'),\n",
        "                           Attrs(Protocol_REF='nucleic acid extraction - standard procedure 2'),\n",
        "                           Attrs(Protocol_REF='library construction'),\n",
        "                           Attrs(Protocol_REF='sequence analysis - standard procedure 7'),\n",
        "                           Attrs(Protocol_REF='reverse transcription - standard procedure 5'),\n",
        "                           Attrs(Protocol_REF='pyrosequencing - standard procedure 6')],\n",
        "          'Raw Data File': ['EVUSNDQ03.sff'],\n",
        "          'Sample Name': ['GSM255772']}\n",
        "       * Node EVNG8PH04.sff Raw Data File\n",
        "         metadata: {'Derived Data File': ['GSE10119_series_matrix.txt'],\n",
        "          'Material Type': [Attrs(Material_Type='mRNA', Term_Source_REF='CHEBI', Term_Accession_Number='16991')],\n",
        "          'Protocol REF': [Attrs(Protocol_REF='mRNA extraction - standard procedure 3'),\n",
        "                           Attrs(Protocol_REF='nucleic acid extraction - standard procedure 2'),\n",
        "                           Attrs(Protocol_REF='library construction'),\n",
        "                           Attrs(Protocol_REF='sequence analysis - standard procedure 7'),\n",
        "                           Attrs(Protocol_REF='reverse transcription - standard procedure 5'),\n",
        "                           Attrs(Protocol_REF='pyrosequencing - standard procedure 6')],\n",
        "          'Raw Data File': ['EVNG8PH04.sff'],\n",
        "          'Sample Name': ['GSM255773']}\n",
        "       * Node EVUSNDQ02.sff Raw Data File\n",
        "         metadata: {'Derived Data File': ['GSE10119_series_matrix.txt'],\n",
        "          'Material Type': [Attrs(Material_Type='mRNA', Term_Source_REF='CHEBI', Term_Accession_Number='16991')],\n",
        "          'Protocol REF': [Attrs(Protocol_REF='mRNA extraction - standard procedure 3'),\n",
        "                           Attrs(Protocol_REF='nucleic acid extraction - standard procedure 2'),\n",
        "                           Attrs(Protocol_REF='library construction'),\n",
        "                           Attrs(Protocol_REF='sequence analysis - standard procedure 7'),\n",
        "                           Attrs(Protocol_REF='reverse transcription - standard procedure 5'),\n",
        "                           Attrs(Protocol_REF='pyrosequencing - standard procedure 6')],\n",
        "          'Raw Data File': ['EVUSNDQ02.sff'],\n",
        "          'Sample Name': ['GSM255771']}\n"
       ]
      }
     ],
     "prompt_number": 17
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [],
     "language": "python",
     "metadata": {},
     "outputs": []
    }
   ],
   "metadata": {}
  }
 ]
}