{
 "metadata": {
  "name": ""
 },
 "nbformat": 3,
 "nbformat_minor": 0,
 "worksheets": [
  {
   "cells": [
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "import sys\n",
      "import os\n",
      "import re\n",
      "import tempfile\n",
      "import itertools\n",
      "import pybedtools\n",
      "import subprocess"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 2
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "#adding the lib path\n",
      "sys.path.append('/Users/abhishek/dev/appys/lib/')"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 3
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "import bamUtils"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 4
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "def bam_to_bigWig(bam,scale=False):\n",
      "    base_path = os.path.dirname(bamFile) or '.'\n",
      "    bamFile_name = os.path.basename(bamFile)\n",
      "    out_bigWig_file = base_path + '/' + bamFile_name.replace('.bam','') + '.bw'\n",
      "    \n",
      "    #constructing genome size file on the fly from bam file\n",
      "    bam_fh,bam_prefix = bamUtils.get_apt_file_handle_for_mapped_file(bamFile)\n",
      "    \n",
      "    temp_genome_file = tempfile.NamedTemporaryFile(mode='w+t',suffix='.tmp', delete=False) #when delete is True file is deleted as soon as it closed\n",
      "    [temp_genome_file.write('%s\\t%s\\n' % (chr,len)) for chr,len in itertools.izip(bam_fh.references,bam_fh.lengths) ]\n",
      "    temp_genome_file.close()\n",
      "    \n",
      "    if scale:\n",
      "        readcount = bamUtils.get_mapped_read_count(bam)\n",
      "        _scale = 1 / (readcount / 1e6)\n",
      "        \n",
      "    \n",
      "    bam_bedtool = pybedtools.BedTool(bamFile)\n",
      "    bedGraph_file = bam_bedtool.genome_coverage(bga=True,g=temp_genome_file.name,scale=_scale)\n",
      "    \n",
      "    #convert to bigwig using the UCSC script\n",
      "    #should be in the path\n",
      "    cmds = ['bedGraphToBigWig', bedGraph_file.fn, str(temp_genome_file.name),out_bigWig_file]\n",
      "    p = subprocess.Popen(cmds,stdout=subprocess.PIPE,stderr=subprocess.PIPE)\n",
      "    stdout,stderr = p.communicate()\n",
      "    \n",
      "    if not stderr:\n",
      "        print 'Created a bigWig file %s for %s' % (os.path.basename(out_bigWig_file),\n",
      "                                                   os.path.basename(bam))\n"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [],
     "prompt_number": 5
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [
      "bamFile = \"/Users/abhishek/testing/datasets/syn2247330_subsampled.bam\"\n",
      "\n",
      "bam_to_bigWig(bamFile,scale=True)"
     ],
     "language": "python",
     "metadata": {},
     "outputs": [
      {
       "ename": "BEDToolsError",
       "evalue": "('Error message from BEDTools written to stderr, above', '\\n*****ERROR: Unrecognized parameter: /var/folders/1n/y92vmpms2llcp2y_36bf_2f40000gp/T/tmpqhvt7A.tmp *****\\n\\n\\nTool:    bedtools genomecov (aka genomeCoverageBed)\\nVersion: v2.17.0\\nSummary: Compute the coverage of a feature file among a genome.\\n\\nUsage: bedtools genomecov [OPTIONS] -i <bed/gff/vcf> -g <genome>\\n\\nOptions: \\n\\t-ibam\\t\\tThe input file is in BAM format.\\n\\t\\t\\tNote: BAM _must_ be sorted by position\\n\\n\\t-d\\t\\tReport the depth at each genome position (with one-based coordinates).\\n\\t\\t\\tDefault behavior is to report a histogram.\\n\\n\\t-dz\\t\\tReport the depth at each genome position (with zero-based coordinates).\\n\\t\\t\\tReports only non-zero positions.\\n\\t\\t\\tDefault behavior is to report a histogram.\\n\\n\\t-bg\\t\\tReport depth in BedGraph format. For details, see:\\n\\t\\t\\tgenome.ucsc.edu/goldenPath/help/bedgraph.html\\n\\n\\t-bga\\t\\tReport depth in BedGraph format, as above (-bg).\\n\\t\\t\\tHowever with this option, regions with zero \\n\\t\\t\\tcoverage are also reported. This allows one to\\n\\t\\t\\tquickly extract all regions of a genome with 0 \\n\\t\\t\\tcoverage by applying: \"grep -w 0$\" to the output.\\n\\n\\t-split\\t\\tTreat \"split\" BAM or BED12 entries as distinct BED intervals.\\n\\t\\t\\twhen computing coverage.\\n\\t\\t\\tFor BAM files, this uses the CIGAR \"N\" and \"D\" operations \\n\\t\\t\\tto infer the blocks for computing coverage.\\n\\t\\t\\tFor BED12 files, this uses the BlockCount, BlockStarts, and BlockEnds\\n\\t\\t\\tfields (i.e., columns 10,11,12).\\n\\n\\t-strand\\t\\tCalculate coverage of intervals from a specific strand.\\n\\t\\t\\tWith BED files, requires at least 6 columns (strand is column 6). \\n\\t\\t\\t- (STRING): can be + or -\\n\\n\\t-5\\t\\tCalculate coverage of 5\" positions (instead of entire interval).\\n\\n\\t-3\\t\\tCalculate coverage of 3\" positions (instead of entire interval).\\n\\n\\t-max\\t\\tCombine all positions with a depth >= max into\\n\\t\\t\\ta single bin in the histogram. Irrelevant\\n\\t\\t\\tfor -d and -bedGraph\\n\\t\\t\\t- (INTEGER)\\n\\n\\t-scale\\t\\tScale the coverage by a constant factor.\\n\\t\\t\\tEach coverage value is multiplied by this factor before being reported.\\n\\t\\t\\tUseful for normalizing coverage by, e.g., reads per million (RPM).\\n\\t\\t\\t- Default is 1.0; i.e., unscaled.\\n\\t\\t\\t- (FLOAT)\\n\\n\\t-trackline\\tAdds a UCSC/Genome-Browser track line definition in the first line of the output.\\n\\t\\t\\t- See here for more details about track line definition:\\n\\t\\t\\t      http://genome.ucsc.edu/goldenPath/help/bedgraph.html\\n\\t\\t\\t- NOTE: When adding a trackline definition, the output BedGraph can be easily\\n\\t\\t\\t      uploaded to the Genome Browser as a custom track,\\n\\t\\t\\t      BUT CAN NOT be converted into a BigWig file (w/o removing the first line).\\n\\n\\t-trackopts\\tWrites additional track line definition parameters in the first line.\\n\\t\\t\\t- Example:\\n\\t\\t\\t   -trackopts \\'name=\"My Track\" visibility=2 color=255,30,30\\'\\n\\t\\t\\t   Note the use of single-quotes if you have spaces in your parameters.\\n\\t\\t\\t- (TEXT)\\n\\nNotes: \\n\\t(1) The genome file should tab delimited and structured as follows:\\n\\t <chromName><TAB><chromSize>\\n\\n\\tFor example, Human (hg19):\\n\\tchr1\\t249250621\\n\\tchr2\\t243199373\\n\\t...\\n\\tchr18_gl000207_random\\t4262\\n\\n\\t(2) The input BED (-i) file must be grouped by chromosome.\\n\\t A simple \"sort -k 1,1 <BED> > <BED>.sorted\" will suffice.\\n\\n\\t(3) The input BAM (-ibam) file must be sorted by position.\\n\\t A \"samtools sort <BAM>\" should suffice.\\n\\nTips: \\n\\tOne can use the UCSC Genome Browser\\'s MySQL database to extract\\n\\tchromosome sizes. For example, H. sapiens:\\n\\n\\tmysql --user=genome --host=genome-mysql.cse.ucsc.edu -A -e \\\\\\n\\t\"select chrom, size from hg19.chromInfo\" > hg19.genome\\n\\n')",
       "output_type": "pyerr",
       "traceback": [
        "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m\n\u001b[0;31mBEDToolsError\u001b[0m                             Traceback (most recent call last)",
        "\u001b[0;32m<ipython-input-7-a4e6da66ac06>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[0mbamFile\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m\"/Users/abhishek/testing/datasets/syn2247330_subsampled.bam\"\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      2\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 3\u001b[0;31m \u001b[0mbam_to_bigWig\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mbamFile\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mscale\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mTrue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
        "\u001b[0;32m<ipython-input-5-4bac38750c65>\u001b[0m in \u001b[0;36mbam_to_bigWig\u001b[0;34m(bam, scale)\u001b[0m\n\u001b[1;32m     17\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     18\u001b[0m     \u001b[0mbam_bedtool\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mpybedtools\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mBedTool\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mbamFile\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 19\u001b[0;31m     \u001b[0mbedGraph_file\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mbam_bedtool\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mgenome_coverage\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mbga\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mTrue\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mg\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mtemp_genome_file\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mname\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mscale\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mscale\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     20\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     21\u001b[0m     \u001b[0;31m#convert to bigwig using the UCSC script\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;32m/Users/abhishek/anaconda/python.app/Contents/lib/python2.7/site-packages/pybedtools/bedtool.pyc\u001b[0m in \u001b[0;36mdecorated\u001b[0;34m(self, *args, **kwargs)\u001b[0m\n\u001b[1;32m    621\u001b[0m             \u001b[0;31m# this calls the actual method in the first place; *result* is\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    622\u001b[0m             \u001b[0;31m# whatever you get back\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 623\u001b[0;31m             \u001b[0mresult\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mmethod\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    624\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    625\u001b[0m             \u001b[0;31m# add appropriate tags\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;32m/Users/abhishek/anaconda/python.app/Contents/lib/python2.7/site-packages/pybedtools/bedtool.pyc\u001b[0m in \u001b[0;36mwrapped\u001b[0;34m(self, *args, **kwargs)\u001b[0m\n\u001b[1;32m    214\u001b[0m             \u001b[0;31m# Do the actual call\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    215\u001b[0m             stream = call_bedtools(cmds, tmp, stdin=stdin,\n\u001b[0;32m--> 216\u001b[0;31m                                    check_stderr=check_stderr)\n\u001b[0m\u001b[1;32m    217\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    218\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0mdoes_not_return_bedtool\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;32m/Users/abhishek/anaconda/python.app/Contents/lib/python2.7/site-packages/pybedtools/helpers.pyc\u001b[0m in \u001b[0;36mcall_bedtools\u001b[0;34m(cmds, tmpfn, stdin, check_stderr)\u001b[0m\n\u001b[1;32m    410\u001b[0m             \u001b[0msys\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mstderr\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mwrite\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mstderr\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    411\u001b[0m             raise BEDToolsError('Error message from BEDTools written to '\n\u001b[0;32m--> 412\u001b[0;31m                                 'stderr, above', stderr)\n\u001b[0m\u001b[1;32m    413\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    414\u001b[0m     \u001b[0;32mexcept\u001b[0m \u001b[0;34m(\u001b[0m\u001b[0mOSError\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mIOError\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0merr\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
        "\u001b[0;31mBEDToolsError\u001b[0m: ('Error message from BEDTools written to stderr, above', '\\n*****ERROR: Unrecognized parameter: /var/folders/1n/y92vmpms2llcp2y_36bf_2f40000gp/T/tmpqhvt7A.tmp *****\\n\\n\\nTool:    bedtools genomecov (aka genomeCoverageBed)\\nVersion: v2.17.0\\nSummary: Compute the coverage of a feature file among a genome.\\n\\nUsage: bedtools genomecov [OPTIONS] -i <bed/gff/vcf> -g <genome>\\n\\nOptions: \\n\\t-ibam\\t\\tThe input file is in BAM format.\\n\\t\\t\\tNote: BAM _must_ be sorted by position\\n\\n\\t-d\\t\\tReport the depth at each genome position (with one-based coordinates).\\n\\t\\t\\tDefault behavior is to report a histogram.\\n\\n\\t-dz\\t\\tReport the depth at each genome position (with zero-based coordinates).\\n\\t\\t\\tReports only non-zero positions.\\n\\t\\t\\tDefault behavior is to report a histogram.\\n\\n\\t-bg\\t\\tReport depth in BedGraph format. For details, see:\\n\\t\\t\\tgenome.ucsc.edu/goldenPath/help/bedgraph.html\\n\\n\\t-bga\\t\\tReport depth in BedGraph format, as above (-bg).\\n\\t\\t\\tHowever with this option, regions with zero \\n\\t\\t\\tcoverage are also reported. This allows one to\\n\\t\\t\\tquickly extract all regions of a genome with 0 \\n\\t\\t\\tcoverage by applying: \"grep -w 0$\" to the output.\\n\\n\\t-split\\t\\tTreat \"split\" BAM or BED12 entries as distinct BED intervals.\\n\\t\\t\\twhen computing coverage.\\n\\t\\t\\tFor BAM files, this uses the CIGAR \"N\" and \"D\" operations \\n\\t\\t\\tto infer the blocks for computing coverage.\\n\\t\\t\\tFor BED12 files, this uses the BlockCount, BlockStarts, and BlockEnds\\n\\t\\t\\tfields (i.e., columns 10,11,12).\\n\\n\\t-strand\\t\\tCalculate coverage of intervals from a specific strand.\\n\\t\\t\\tWith BED files, requires at least 6 columns (strand is column 6). \\n\\t\\t\\t- (STRING): can be + or -\\n\\n\\t-5\\t\\tCalculate coverage of 5\" positions (instead of entire interval).\\n\\n\\t-3\\t\\tCalculate coverage of 3\" positions (instead of entire interval).\\n\\n\\t-max\\t\\tCombine all positions with a depth >= max into\\n\\t\\t\\ta single bin in the histogram. Irrelevant\\n\\t\\t\\tfor -d and -bedGraph\\n\\t\\t\\t- (INTEGER)\\n\\n\\t-scale\\t\\tScale the coverage by a constant factor.\\n\\t\\t\\tEach coverage value is multiplied by this factor before being reported.\\n\\t\\t\\tUseful for normalizing coverage by, e.g., reads per million (RPM).\\n\\t\\t\\t- Default is 1.0; i.e., unscaled.\\n\\t\\t\\t- (FLOAT)\\n\\n\\t-trackline\\tAdds a UCSC/Genome-Browser track line definition in the first line of the output.\\n\\t\\t\\t- See here for more details about track line definition:\\n\\t\\t\\t      http://genome.ucsc.edu/goldenPath/help/bedgraph.html\\n\\t\\t\\t- NOTE: When adding a trackline definition, the output BedGraph can be easily\\n\\t\\t\\t      uploaded to the Genome Browser as a custom track,\\n\\t\\t\\t      BUT CAN NOT be converted into a BigWig file (w/o removing the first line).\\n\\n\\t-trackopts\\tWrites additional track line definition parameters in the first line.\\n\\t\\t\\t- Example:\\n\\t\\t\\t   -trackopts \\'name=\"My Track\" visibility=2 color=255,30,30\\'\\n\\t\\t\\t   Note the use of single-quotes if you have spaces in your parameters.\\n\\t\\t\\t- (TEXT)\\n\\nNotes: \\n\\t(1) The genome file should tab delimited and structured as follows:\\n\\t <chromName><TAB><chromSize>\\n\\n\\tFor example, Human (hg19):\\n\\tchr1\\t249250621\\n\\tchr2\\t243199373\\n\\t...\\n\\tchr18_gl000207_random\\t4262\\n\\n\\t(2) The input BED (-i) file must be grouped by chromosome.\\n\\t A simple \"sort -k 1,1 <BED> > <BED>.sorted\" will suffice.\\n\\n\\t(3) The input BAM (-ibam) file must be sorted by position.\\n\\t A \"samtools sort <BAM>\" should suffice.\\n\\nTips: \\n\\tOne can use the UCSC Genome Browser\\'s MySQL database to extract\\n\\tchromosome sizes. For example, H. sapiens:\\n\\n\\tmysql --user=genome --host=genome-mysql.cse.ucsc.edu -A -e \\\\\\n\\t\"select chrom, size from hg19.chromInfo\" > hg19.genome\\n\\n')"
       ]
      },
      {
       "output_type": "stream",
       "stream": "stderr",
       "text": [
        "\n",
        "Command was:\n",
        "\n",
        "\tbedtools genomecov -bga -scale -g /var/folders/1n/y92vmpms2llcp2y_36bf_2f40000gp/T/tmpqhvt7A.tmp -ibam /Users/abhishek/testing/datasets/syn2247330_subsampled.bam\n",
        "\n",
        "Error message was:\n",
        "\n",
        "*****ERROR: Unrecognized parameter: /var/folders/1n/y92vmpms2llcp2y_36bf_2f40000gp/T/tmpqhvt7A.tmp *****\n",
        "\n",
        "\n",
        "Tool:    bedtools genomecov (aka genomeCoverageBed)\n",
        "Version: v2.17.0\n",
        "Summary: Compute the coverage of a feature file among a genome.\n",
        "\n",
        "Usage: bedtools genomecov [OPTIONS] -i <bed/gff/vcf> -g <genome>\n",
        "\n",
        "Options: \n",
        "\t-ibam\t\tThe input file is in BAM format.\n",
        "\t\t\tNote: BAM _must_ be sorted by position\n",
        "\n",
        "\t-d\t\tReport the depth at each genome position (with one-based coordinates).\n",
        "\t\t\tDefault behavior is to report a histogram.\n",
        "\n",
        "\t-dz\t\tReport the depth at each genome position (with zero-based coordinates).\n",
        "\t\t\tReports only non-zero positions.\n",
        "\t\t\tDefault behavior is to report a histogram.\n",
        "\n",
        "\t-bg\t\tReport depth in BedGraph format. For details, see:\n",
        "\t\t\tgenome.ucsc.edu/goldenPath/help/bedgraph.html\n",
        "\n",
        "\t-bga\t\tReport depth in BedGraph format, as above (-bg).\n",
        "\t\t\tHowever with this option, regions with zero \n",
        "\t\t\tcoverage are also reported. This allows one to\n",
        "\t\t\tquickly extract all regions of a genome with 0 \n",
        "\t\t\tcoverage by applying: \"grep -w 0$\" to the output.\n",
        "\n",
        "\t-split\t\tTreat \"split\" BAM or BED12 entries as distinct BED intervals.\n",
        "\t\t\twhen computing coverage.\n",
        "\t\t\tFor BAM files, this uses the CIGAR \"N\" and \"D\" operations \n",
        "\t\t\tto infer the blocks for computing coverage.\n",
        "\t\t\tFor BED12 files, this uses the BlockCount, BlockStarts, and BlockEnds\n",
        "\t\t\tfields (i.e., columns 10,11,12).\n",
        "\n",
        "\t-strand\t\tCalculate coverage of intervals from a specific strand.\n",
        "\t\t\tWith BED files, requires at least 6 columns (strand is column 6). \n",
        "\t\t\t- (STRING): can be + or -\n",
        "\n",
        "\t-5\t\tCalculate coverage of 5\" positions (instead of entire interval).\n",
        "\n",
        "\t-3\t\tCalculate coverage of 3\" positions (instead of entire interval).\n",
        "\n",
        "\t-max\t\tCombine all positions with a depth >= max into\n",
        "\t\t\ta single bin in the histogram. Irrelevant\n",
        "\t\t\tfor -d and -bedGraph\n",
        "\t\t\t- (INTEGER)\n",
        "\n",
        "\t-scale\t\tScale the coverage by a constant factor.\n",
        "\t\t\tEach coverage value is multiplied by this factor before being reported.\n",
        "\t\t\tUseful for normalizing coverage by, e.g., reads per million (RPM).\n",
        "\t\t\t- Default is 1.0; i.e., unscaled.\n",
        "\t\t\t- (FLOAT)\n",
        "\n",
        "\t-trackline\tAdds a UCSC/Genome-Browser track line definition in the first line of the output.\n",
        "\t\t\t- See here for more details about track line definition:\n",
        "\t\t\t      http://genome.ucsc.edu/goldenPath/help/bedgraph.html\n",
        "\t\t\t- NOTE: When adding a trackline definition, the output BedGraph can be easily\n",
        "\t\t\t      uploaded to the Genome Browser as a custom track,\n",
        "\t\t\t      BUT CAN NOT be converted into a BigWig file (w/o removing the first line).\n",
        "\n",
        "\t-trackopts\tWrites additional track line definition parameters in the first line.\n",
        "\t\t\t- Example:\n",
        "\t\t\t   -trackopts 'name=\"My Track\" visibility=2 color=255,30,30'\n",
        "\t\t\t   Note the use of single-quotes if you have spaces in your parameters.\n",
        "\t\t\t- (TEXT)\n",
        "\n",
        "Notes: \n",
        "\t(1) The genome file should tab delimited and structured as follows:\n",
        "\t <chromName><TAB><chromSize>\n",
        "\n",
        "\tFor example, Human (hg19):\n",
        "\tchr1\t249250621\n",
        "\tchr2\t243199373\n",
        "\t...\n",
        "\tchr18_gl000207_random\t4262\n",
        "\n",
        "\t(2) The input BED (-i) file must be grouped by chromosome.\n",
        "\t A simple \"sort -k 1,1 <BED> > <BED>.sorted\" will suffice.\n",
        "\n",
        "\t(3) The input BAM (-ibam) file must be sorted by position.\n",
        "\t A \"samtools sort <BAM>\" should suffice.\n",
        "\n",
        "Tips: \n",
        "\tOne can use the UCSC Genome Browser's MySQL database to extract\n",
        "\tchromosome sizes. For example, H. sapiens:\n",
        "\n",
        "\tmysql --user=genome --host=genome-mysql.cse.ucsc.edu -A -e \\\n",
        "\t\"select chrom, size from hg19.chromInfo\" > hg19.genome\n",
        "\n"
       ]
      }
     ],
     "prompt_number": 7
    },
    {
     "cell_type": "code",
     "collapsed": false,
     "input": [],
     "language": "python",
     "metadata": {},
     "outputs": []
    }
   ],
   "metadata": {}
  }
 ]
}